torch/csrc/autograd/engine.cpp

```diff
@@ -543,7 +543,12 @@ void validate_outputs(
       }
       grads[i] = at::sum_to(std::move(grads[i]), metadata.shape());
     }
-    TORCH_CHECK(isFloatingType(grads[i].scalar_type()));
+
+    bool input_is_complex = isComplexType(c10::typeMetaToScalarType(metadata.options().dtype()));
+    bool grad_is_complex = isComplexType(grads[i].scalar_type());
+
+    TORCH_CHECK(isFloatingType(grads[i].scalar_type()) || ((input_is_complex && grad_is_complex)
```

Comment:
For the complex half, you simply want `input_is_complex == grad_is_complex` no?

Reply:
we currently don't support complex half

Reply:
oh you meant the code related to complex and not the dtype complex half, I see haha. yup makes sense, updated!